External/Plugins/ASCompletion/PluginMain.cs

```diff
@@ -446,104 +446,107 @@ public void HandleEvent(Object sender, NotifyEvent e, HandlingPriority priority)
-                            {
-                                ASComplete.OnShortcut(Keys.Control | Keys.Shift | Keys.Space, ASContext.CurSciControl);
-                                e.Handled = true;
-                            }
-                            else if (command == "ASCompletion.CtrlAltSpace")
-                            {
-                                ASComplete.OnShortcut(Keys.Control | Keys.Alt | Keys.Space, ASContext.CurSciControl);
-                                e.Handled = true;
-                            }
-                            else if (command == "ASCompletion.ContextualGenerator")
-                            {
-                                if (ASContext.HasContext && ASContext.Context.IsFileValid)
+                                // resolve element under cursor and send a CustomData event
+                                else if (command == "ASCompletion.ResolveElement")
+                                {
+                                    ASComplete.ResolveElement(sci, cmdData);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.MakeIntrinsic")
+                                {
+                                    ASContext.Context.MakeIntrinsic(cmdData);
+                                    e.Handled = true;
+                                }
+
+                                // alternative to default shortcuts
+                                else if (command == "ASCompletion.CtrlSpace")
+                                {
+                                    ASComplete.OnShortcut(Keys.Control | Keys.Space, ASContext.CurSciControl);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.CtrlShiftSpace")
                                 {
-                                    ASGenerator.ContextualGenerator(ASContext.CurSciControl);
+                                    ASComplete.OnShortcut(Keys.Control | Keys.Shift | Keys.Space, ASContext.CurSciControl);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.CtrlAltSpace")
+                                {
+                                    ASComplete.OnShortcut(Keys.Control | Keys.Alt | Keys.Space, ASContext.CurSciControl);
+                                    e.Handled = true;
+                                }
+                                else if (command == "ASCompletion.ContextualGenerator")
+                                {
+                                    if (ASContext.HasContext && ASContext.Context.IsFileValid)
+                                    {
+                                        ASGenerator.ContextualGenerator(ASContext.CurSciControl);
```

Comment:
Might be better to avoid making `KnownList` public (although refactoring to use a static prop is fine) and instead doing something like:
```
var options = ASGenerator.ContextualGenerator(ASContext.CurSciControl);
options.AddRange(GetRefactorMenus(de.Data as ToolStripMenuItem));
CompletionList.Show(options, false);
```